packages/opentelemetry-core/src/context/propagation/B3Propagator.ts

```diff
@@ -46,17 +50,27 @@ export class B3Propagator implements HttpTextPropagator {
   inject(context: Context, carrier: unknown, setter: SetterFunction) {
     const spanContext = getParentSpanContext(context);
     if (!spanContext) return;
-
+    const parentSpanId = context.getValue(PARENT_SPAN_ID_KEY);
     if (
       isValidTraceId(spanContext.traceId) &&
       isValidSpanId(spanContext.spanId)
     ) {
+      if (parentSpanId) {
+        if (isValidTraceId(parentSpanId as string))
+          setter(carrier, X_B3_PARENT_SPAN_ID, parentSpanId);
+        else return;
+      }
+      const debug = context.getValue(DEBUG_FLAG_KEY);
       setter(carrier, X_B3_TRACE_ID, spanContext.traceId);
       setter(carrier, X_B3_SPAN_ID, spanContext.spanId);
-
+      // According to the B3 spec, if the debug flag is set,
+      // the sampled flag shouldn't be propagated as well.
+      if (debug === '1') {
+        setter(carrier, X_B3_FLAGS, debug);
+      }
       // We set the header only if there is an existing sampling decision.
       // Otherwise we will omit it => Absent.
-      if (spanContext.traceFlags !== undefined) {
+      else if (spanContext.traceFlags !== undefined) {
```

Comment:
having the else on a separate line here makes it easy to miss. Would rather have it on the same line as the if block termination bracket and move the comment inside the else block.

Reply:
Will do